pallets/pallet-asset-switch/src/benchmarking.rs

```diff
@@ -0,0 +1,236 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use xcm::VersionedMultiAsset;
+
+pub struct BenchmarkInfo {
+	pub remote_fee: VersionedMultiAsset,
+}
+
+/// Helper trait implementable by the runtime to set some additional state
+/// before the pallet benchmarks are run.
+///
+/// This is highly dependent on the runtime logic.
+/// If no special conditions are to be met, it can simply be a no-op and return
+/// `None`.
+pub trait BenchmarkHelper {
+	fn setup() -> Option<BenchmarkInfo>;
+}
+
+impl BenchmarkHelper for () {
+	fn setup() -> Option<BenchmarkInfo> {
+		None
+	}
+}
+
+#[frame_benchmarking::v2::instance_benchmarks(where LocalCurrencyBalanceOf<T, I>: Into<u128>)]
+mod benchmarks {
+
+	use frame_support::traits::{
+		fungible::{Inspect as InspectFungible, Mutate as MutateFungible},
+		EnsureOrigin,
+	};
+	use frame_system::RawOrigin;
+	use sp_runtime::traits::{TryConvert, Zero};
+	use sp_std::{boxed::Box, vec};
+	use xcm::v3::{AssetId, Fungibility, Junction, Junctions, MultiAsset, MultiLocation, XcmContext};
+	use xcm_executor::traits::TransactAsset;
+
+	use crate::{
+		benchmarking::{BenchmarkHelper, BenchmarkInfo},
+		Call, Config, LocalCurrencyBalanceOf, Pallet, SwitchPairStatus,
+	};
+
+	const RESERVE_LOCATION: MultiLocation = MultiLocation {
+		parents: 1,
+		interior: Junctions::X1(Junction::Parachain(1_000)),
+	};
+	const REMOTE_ASSET_ID: AssetId = AssetId::Concrete(RESERVE_LOCATION);
+	const REMOTE_FEE: MultiAsset = MultiAsset {
+		id: REMOTE_ASSET_ID,
+		fun: Fungibility::Fungible(100_000),
```

Comment:
Not sure, but I would probably prefer to have those const values in the default implementation of the `BenchmarkHelper`

Reply:
But the purpose of these defaults is if the benchmark helper does not want to have anything to do with the setup, which would return `None`. In that case we assume "anything will do", and we set a default ourselves. Alternatively we could force the helper to return a value, but then we can't use `()` without attaching some more assumptions to it.

Reply:
See comment below: https://github.com/KILTprotocol/kilt-node/pull/671#discussion_r1684460262